### PR TITLE
Update washing machine to be able to use clean water.

### LINKED ISSUE #23697: Luxury RV lacks a default (dirty) water tank for the washing machine
Game version: 7637

Operating system: Win7

Tiles or curses: Tiles

Mods active:
I think Folding Parts might be the only one that touches vehicles?
### Expected behavior
Find a Luxury RV (with semi-working solar panels and a charged battery), dump a bunch of dirty clothes in the washing machine, turn it on. I would assume it uses clean water from the clean water tank??
### Actual behavior
Washing machine demands water from a water tank ("24 charges from a vehicle tank"). Boil a bunch of water to top off my clean water tank. Washing machine demands water from a water tank anyway. Wish for a plastic jerrycan and weld it under the Lux RV. Fill it with regular unfiltered water. Washing machine accepts the offering of water and turns on.

### PR DESCRIPTION
**Pull request by laeaton2001**
Allows washing machines to use clean water, it will default to dirty water if enough available.

Closes #23697

### CODE REVIEW

**Comment by laeaton2001** [comment]
Been a while since I contributed, got to get my styling settings correct again.

File: src/vehicle.cpp [hunk 1]
```diff
@@ -6545,7 +6545,12 @@ void vehicle::use_washing_machine( int p ) {
             n.set_age( 0 );
         }
 
-        drain( "water", 24 );
+        if ( fuel_left( "water" ) >= 24 ) {
+            drain( "water", 24 );
+        }
+        else {
+            drain( "water_clean", 24 );
```
Comment (ZhilkinSerg) [inline src/vehicle.cpp]: What if there is less than 24 charges of  `water_clean` here?

Comment (Night-Pryanik) [reply src/vehicle.cpp]: For now this means that if there are less than 24 units of ordinary `water`, it drains water clean, no matter if there is clean water at all. There should be check for clean water, not just simple `else`.
Also, styling: should be `} else {`.

File: src/vehicle.cpp [hunk 2]
```diff
@@ -6545,7 +6545,12 @@ void vehicle::use_washing_machine( int p ) {
             n.set_age( 0 );
         }
 
-        drain( "water", 24 );
+        if ( fuel_left( "water" ) >= 24 ) {
```
Comment (Night-Pryanik) [inline src/vehicle.cpp]: `if` should be glued to the opening bracket.

File: src/vehicle.cpp (hunk 1 above)
Comment (laeaton2001) [reply src/vehicle.cpp]: The check for water amounts is done prior, this code can't be executed unless there is either more than 24 charges of water or clean water (or both). If there is less the "you need 24 charges of water" message is displayed. Or am I missing something?

File: src/vehicle.cpp (hunk 2 above)
Comment (laeaton2001) [reply src/vehicle.cpp]: Will fix this today.

**Pull request merged**